File: src/Training.FileExplorer.ClientApp/src/infrastructure/models/entities/StorageDrive.ts
import { StorageEntryType } from "@/infrastructure/models/entities/StorageItemType";
import type { IStorageEntry } from "@/infrastructure/models/entities/IStorageEntry";

export class StorageDrive implements IStorageEntry {
    name!: string;
    label!: string;
    format!: string;
    type!: string;
    totalSpace!: number;
    freeSpace!: number;
    unavailableSpace!: number;
    usedSpace!: number;
    entryType!: StorageEntryType;
    path!: string;
}

File: src/FromScratch.ServerApp/FileExplorer.Application/FileStorage/Models/StorageDrive.cs
namespace FileExplorer.Application.FileStorage.Models;

public class StorageDrive : IStorageEntry
{
    public string Name { get; set; } = string.Empty; // C

    public string Path { get; set; } = string.Empty; // C:\\

    public string Format { get; set; } = string.Empty; // NTFS, Fat32

    public string Type { get; set; } = string.Empty;

    public long TotalSpace { get; set; }

    public long FreeSpace { get; set; }

    public long UnavailableSpace { get; set; }

    public long UsedSpace { get; set; }

    public StorageEntryType EntryType { get; set; } = StorageEntryType.Drive;
}